engines/ultima/ultima8/audio/music_process.cpp

```diff
@@ -96,9 +88,23 @@ void MusicProcess::unqueueMusic() {
 
 void MusicProcess::restoreMusic() {
 	_queuedTrack = 0;
+	_combatMusicActive = false;
 	playMusic_internal(_lastRequest);
 }
 
+void MusicProcess::getTrackState(int trackState[3]) const {
+	trackState[0] = _wantedTrack;
+	trackState[1] = _lastRequest;
+	trackState[2] = _queuedTrack;
```

Comment:
If I'm honest this feels a little hacky, I considered creating a struct to make it cleaner.  Happy to take input on it.

Reply:
I agree. A structure would be better. You could also have the structure have a default constructor to initialize the fields. Also, Common::fill should be used in preference to direct memset calls generally, though in this case neither will be needed if you make the structure.

Reply:
Thanks. I updated it to be a struct.  Feels better.